python/orca/src/bigdl/orca/ray/ray_on_spark_context.py

```diff
@@ -527,22 +546,22 @@ def purge(self):
                 self.ray_service.gen_stop()).collect()
         self.initialized = False
 
-    def _get_spark_local_cores(self):
-        local_symbol = re.match(r"local\[(.*)\]", self.sc.master).group(1)
+    def _get_spark_local_cores(self) -> int:
+        local_symbol = re.match(r"local\[(.*)\]", self.sc.master).group(1)  # type: ignore
         if local_symbol == "*":
             return multiprocessing.cpu_count()
         else:
             return int(local_symbol)
 
-    def _update_extra_params(self, extra_params):
+    def _update_extra_params(self, extra_params: Optional[Dict[str, str]]) -> Dict[str, str]:
         kwargs = {}
         if extra_params is not None:
             for k, v in extra_params.items():
                 kw = k.replace("-", "_")
                 kwargs[kw] = v
         return kwargs
 
-    def init(self, driver_cores=0):
+    def init(self, driver_cores: int = 0):
```

Comment:
return a dict[str, str]?

Reply:
Not necessarily. The return from ray is complicated and ray itself doesn't hint the return type. So we also ignore here.